pkg/util/v1beta1/katibconfig/config_test.go

```diff
@@ -0,0 +1,455 @@
+	fakeClientBuilder := fake.NewClientBuilder().WithScheme(scheme.Scheme)
+	if katibConfigMap != nil {
+		fakeClientBuilder.WithObjects(katibConfigMap)
+	}
+	return fakeClientBuilder.Build()
+}
+
+func newFakeKatibConfigMap(config *katibConfig) *corev1.ConfigMap {
+
+	data := map[string]string{}
+	if config.suggestion != nil {
+		bSuggestionConfig, _ := json.Marshal(config.suggestion)
+		data[consts.LabelSuggestionTag] = string(bSuggestionConfig)
+	}
+	if config.earlyStopping != nil {
+		bEarlyStoppingConfig, _ := json.Marshal(config.earlyStopping)
+		data[consts.LabelEarlyStoppingTag] = string(bEarlyStoppingConfig)
+	}
+	if config.metricsCollector != nil {
+		bMetricsCollector, _ := json.Marshal(config.metricsCollector)
+		data[consts.LabelMetricsCollectorSidecar] = string(bMetricsCollector)
+	}
+
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      consts.KatibConfigMapName,
+			Namespace: consts.DefaultKatibNamespace,
+		},
+		Data: data,
+	}
+}
+
+func newFakeSuggestionConfig() *SuggestionConfig {
+	defaultVolumeStorage, _ := resource.ParseQuantity(consts.DefaultSuggestionVolumeStorage)
+
+	return &SuggestionConfig{
+		Image:           "suggestion-image",
+		ImagePullPolicy: consts.DefaultImagePullPolicy,
+		Resource:        *setFakeResourceRequirements(),
+		VolumeMountPath: consts.DefaultContainerSuggestionVolumeMountPath,
+		PersistentVolumeClaimSpec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				consts.DefaultSuggestionVolumeAccessMode,
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: map[corev1.ResourceName]resource.Quantity{
+					corev1.ResourceStorage: defaultVolumeStorage,
+				},
+			},
+		},
+		PersistentVolumeSpec: corev1.PersistentVolumeSpec{
+			PersistentVolumeReclaimPolicy: corev1.PersistentVolumeReclaimDelete,
+		},
+	}
+}
+
+func newFakeEarlyStoppingConfig() *EarlyStoppingConfig {
+	return &EarlyStoppingConfig{
+		Image:           "early-stopping-image",
+		ImagePullPolicy: consts.DefaultImagePullPolicy,
+	}
+}
+
+func newFakeMetricsCollectorConfig() *MetricsCollectorConfig {
+	return &MetricsCollectorConfig{
+		Image:           "metrics-collector-image",
+		ImagePullPolicy: consts.DefaultImagePullPolicy,
+		Resource:        *setFakeResourceRequirements(),
+	}
+}
+
+func setFakeResourceRequirements() *corev1.ResourceRequirements {
+	defaultCPURequest, _ := resource.ParseQuantity(consts.DefaultCPURequest)
+	defaultMemoryRequest, _ := resource.ParseQuantity(consts.DefaultMemRequest)
+	defaultEphemeralStorageRequest, _ := resource.ParseQuantity(consts.DefaultDiskRequest)
+
+	defaultCPULimit, _ := resource.ParseQuantity(consts.DefaultCPULimit)
+	defaultMemoryLimit, _ := resource.ParseQuantity(consts.DefaultMemLimit)
+	defaultEphemeralStorageLimit, _ := resource.ParseQuantity(consts.DefaultDiskLimit)
```

Comment:
Do you mean what we should use the setResourceRequirements function?

Reply:
I mean set some different value in `resource.ParseQuantity` other than `consts.DefaultCPURequest`, etc.
For example, `defaultCPURequest, _ := resource.ParseQuantity("250m")`

Reply:
I see, thanks.

I specify default value in`setFakeResourceRequirements` function since `setResourceRequirements` function only judge that resources are empty or less than zero.

https://github.com/kubeflow/katib/blob/29409198ff7a13d1984e48a0b14d6954296950d6/pkg/util/v1beta1/katibconfig/config.go#L251

https://github.com/kubeflow/katib/blob/29409198ff7a13d1984e48a0b14d6954296950d6/pkg/util/v1beta1/katibconfig/config.go#L289

Am I missing any other conditions?

Reply:
I think only that two.
So what do you think about settings some values in your `setFakeResourceRequirements` function ?
For example:
```
	defaultCPURequest, _ := resource.ParseQuantity("25m")
	defaultMemoryRequest, _ := resource.ParseQuantity("200Mi")
	defaultEphemeralStorageRequest, _ := resource.ParseQuantity("550Mi")
```

Reply:
It sounds good.
I'd like to fix that.